src/core/index.js

```diff
@@ -18,25 +20,45 @@ function AppBase(config) {
     path = '',
     protocol = '',
   } = URL(config.url || '');
-  let url = host + path;
+  let { url } = config;
+
+  const backendName = backendAlias[config.mongodb ? 'MONGODB' : 'ELASTICSEARCH'];
+  // eslint-disable-next-line
+  const schema = SCHEMA[backendName];
+  validateSchema(
+    {
+      url: config.url,
+      app: config.app,
+      credentials: config.credentials,
+      username: config.username,
+      password: config.password,
+      enableTelemetry: config.enableTelemetry,
+      mongodb: config.mongodb,
+    },
+    schema,
+    backendName,
+  );
+
 
-  // Validate config and throw appropriate error
-  if (typeof url !== 'string' || url === '') {
-    throw new Error('URL not present in options.');
-  }
-  if (typeof config.app !== 'string' || config.app === '') {
-    throw new Error('App name is not present in options.');
-  }
   if (typeof protocol !== 'string' || protocol === '') {
     throw new Error(
       'Protocol is not present in url. URL should be of the form https://appbase-demo-ansible-abxiydt-arc.searchbase.io',
     );
   }
-  // Parse url
-  if (url.slice(-1) === '/') {
-    url = url.slice(0, -1);
-  }
+
   let credentials = auth || null;
+  if (!config.mongodb) {
+    url = host + path;
+    // Parse url
+    if (url.slice(-1) === '/') {
```

Comment:
URL parsing should happen irrespective of search BE.

Reply:
Fixed here 👉  https://github.com/appbaseio/appbase-js/pull/84/commits/314651a097b981e8fef8bf9cb083dcfcb3faafa6